### DISCUSSION TITLE
principle

### DISCUSSION THREAD

**Discussion by MQYForverT** [nuejs/nue#89] [Q&A] [answered]
Hello author, can I understand the working principle of Nue? Including but not limited to compile time, runtime, build process, build products, etc., this will give us a clearer understanding of your design ideas

**Comment by tipiirai** [comment, marked as answer]
Here is the big picture, the roadmap and rough ETA's:

![roadmap-big](https://github.com/nuejs/nuejs/assets/169330/37cc4206-5773-4482-8a8c-853c7b450b2c)

Nuekit is the central binary to be executed, and it uses the other modules like Nue js, Nue CSS, and Nuemark. At first phase the focus is on multi-page applications (aka "websites") so Nuekit acts mainly as a static website generator. It's a JavaScript program that works under Bun and Node. JS/CSS/Markdown is transpiled to a working website that is run on the browser. The idea is to work as close as possible to core web standards so things like Tailwind and TypeScript are a no-go. 

Not sure if I answered your questions. Please be more specific so I can perhaps answer better.
